### PR TITLE
Mook village and basic mook refactor

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
refactors mooks into basic mooks and re-adds them to the game

## Why It's Good For The Game
this refactors mooks into basic mobs and re adds them to the game. mooks are now a part of lavaland. they come as a part of a random ruin which consist of a entire village of friendly mooks. Mooks will aid players but they will still attack ashwalkers because of some troubled history between them.
mookvillage

mooks are a very diseased specie. nanotrasen discovered a small tribe of mooks and cut a deal with their tribal chief to aid ss13 miners in exchange for medical supplies. 

tribal chief in his decked out home
tribalchief

male mooks go out and mine and haul ore off back to their village. they will deposit ores into a stand which is managed by another mook. they will all return to their village to rest once a lavastorm comes.
mookstand

players can use this stand to withdraw any ore they like
mookui

female mooks will stay behind in the village to guard it from ashwalkers. they will also heal male mooks when they come back from a long day of work. the tribal chief is a bum and chooses not to go out and mine. he will stay behind in the village and issue commands to his people rather than work

the village also has its own bard! he follows player visitors and plays nice music for them while they are in the village (although he is not very talented).
mookbard

he is still a warrior at heart tho so he will be smashing his guitar over ashwalker skull
mookshmash

## Changelog
:cl:
refactor: mooks are now basic mobs. please report any bugs
feature: added mook village to lavaland ruins!
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
now we know where they went after kilostation was decomissioned

**Comment by nikothedude** [comment]
Oh my God this is so goofy and cool I love it

**Comment by ArcaneMusic** [comment]
Critically based

**Comment by HoodMilk** [comment]
we NEED this

File: tgui/packages/tgui/interfaces/MaterialStand.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  const [searchItem, setSearchItem] = useLocalState(context, 'searchItem', '');
+  const search = createSearch(searchItem, (ores) => ores.name);
+  const ores_filtered =
+    searchItem.length > 0 ? ores.filter((ore) => search(ore)) : ores;
+  return (
+    <Window title="MaterialStand" width={550} height={400}>
+      <Window.Content>
+        <Stack fill vertical>
+          <Section>
+            <Stack.Item>
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/MaterialStand.tsx:34]: These are in the wrong order, stack items should be the direct descendant

File: tgui/packages/tgui/interfaces/MaterialStand.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                onInput={(e, value) => {
+                  setSearchItem(value);
+                }}
+                fluid
+              />
+            </Stack.Item>
+          </Section>
+          <Section title="Stock" fill scrollable>
+            <Stack.Item>
+              <Flex wrap>
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/MaterialStand.tsx:53]: Why the change to flex here? Is it just for wrap? I think this is on stack if so

File: tgui/packages/tgui/interfaces/MaterialStand.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          </Section>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const RetrieveIcon = (props, context) => {
+  const { data } = useBackend<Data>(context);
+  const { ore_images } = data;
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/MaterialStand.tsx:89]:
```suggestion
  const { ore_images = [] } = data;
```
I don't trust byond to send this stuff

**Review by jlsnow301 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/basic/lavaland/mook/mook.dm [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+	name = "wanderer"
+	desc = "This unhealthy looking primitive is wielding a rudimentary hatchet, swinging it with wild abandon. One isn't much of a threat, but in numbers they can quickly overwhelm a superior opponent."
+	icon = 'icons/mob/simple/jungle/mook.dmi'
+	icon_state = "mook"
+	icon_living = "mook"
+	icon_dead = "mook_dead"
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	gender = MALE
+	maxHealth = 150
+	faction = list("neutral", "mining")
```
Comment (Profakos) [inline code/modules/mob/living/basic/lavaland/mook/mook.dm:18]: Factions have defines now, please use them.

**Comment by nikothedude** [comment]
Isnt there an ash walker faction? Why not check for that faction rather than just checking species? There may come a time where ash walkers can have mobs that arent ash walkers, but still fight for them.

**Comment by chemistrymain2** [comment]
ANOTHER BEN10OMNITRIX W

**Comment by Profakos** [comment]
This is beautiful work, well done!

**Comment by carlarctg** [comment]
This is the greatest refactor yet

**Comment by IndieanaJones** [comment]
The king of soul strikes again with his strongest blow yet

**Comment by SgtHunk** [comment]
Ben10Omintrix doesn't miss 🔥🔥amazing pr

File: code/modules/mob/living/basic/lavaland/mook/mook_ai.dm [hunk 5]
```diff
@@ -0,0 +1,315 @@
+///commands the chief can pick from
+GLOBAL_LIST_INIT(mook_commands, list(
+	new /datum/pet_command/point_targetting/attack(),
+	new /datum/pet_command/point_targetting/fetch(),
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook_ai.dm:4]:
```suggestion
	new /datum/pet_command/point_targetting/attack,
	new /datum/pet_command/point_targetting/fetch,
```

File: code/modules/mob/living/basic/lavaland/mook/mook_ai.dm [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+	var/storm_approaching = controller.blackboard[BB_STORM_APPROACHING]
+	var/mob/living/living_pawn = controller.pawn
+	var/obj/effect/home = controller.blackboard[BB_HOME_VILLAGE]
+	if(QDELETED(home))
+		return
+	if(get_dist(living_pawn, home) < controller.blackboard[BB_MAXIMUM_DISTANCE_TO_VILLAGE])
+		return
+	if(storm_approaching || (locate(/obj/item/stack/ore) in living_pawn))
+		controller.clear_blackboard_key(BB_TARGET_MINERAL_WALL)
+		return ..()
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook_ai.dm:52]: make this an early return instead for clarity because i always think it's weird for a "conditional" call of parent

```suggestion
	if(!storm_approaching || !(locate(/obj/item/stack/ore) in living_pawn))
		return

	controller.clear_blackboard_key(BB_TARGET_MINERAL_WALL)
	return ..()
```

takes up some more space but feels way more logical

File: code/datums/ai/basic_mobs/basic_subtrees/mine_walls.dm [hunk 7]
```diff
@@ -0,0 +1,73 @@
+//behavior to find mineable mineral walls
+
+/datum/ai_planning_subtree/mine_walls
+	var/find_wall_behavior = /datum/ai_behavior/find_mineral_wall
```
Comment (san7890) [inline code/datums/ai/basic_mobs/basic_subtrees/mine_walls.dm:4]: Would `/datum/element/wall_smasher` be considered a required operational datum or do you think this could work without it?

File: code/modules/mob/living/basic/lavaland/mook/mook.dm [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+	var/mutable_appearance/chief_active
+	///overlay in our warmup state
+	var/mutable_appearance/chief_warmup
+
+/mob/living/basic/mining/mook/worker/tribal_chief/Initialize(mapload)
+	. = ..()
+	chief_neutral = mutable_appearance(icon, "mook_chief")
+	chief_strike = mutable_appearance(icon, "mook_chief_strike")
+	chief_active = mutable_appearance(icon, "mook_chief_leap")
+	chief_warmup = mutable_appearance(icon, "mook_chief_warmup")
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook.dm:243]: nah we can do it like this

```suggestion
	///overlay in our neutral state
	var/static/mutable_appearance/chief_neutral = mutable_appearance(icon, "mook_chief")
	///overlay in our striking state
	var/static/mutable_appearance/chief_strike = mutable_appearance(icon, "mook_chief_strike")
	///overlay in our active state
	var/static/mutable_appearance/chief_active = mutable_appearance(icon, "mook_chief_leap")
	///overlay in our warmup state
	var/static/mutable_appearance/chief_warmup = mutable_appearance(icon, "mook_chief_warmup")

/mob/living/basic/mining/mook/worker/tribal_chief/Initialize(mapload)
	. = ..()
```

tada. if you don't believe this works look at `/datum/component/wet_floor`.

File: code/modules/mob/living/basic/lavaland/mook/mook.dm [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+		if(MOOK_ATTACK_NEUTRAL)
+			. += chief_neutral
+		if(MOOK_ATTACK_WARMUP)
+			. += chief_warmup
+		if(MOOK_ATTACK_ACTIVE)
+			. += chief_active
+		if(MOOK_ATTACK_STRIKE)
+			. += chief_strike
+
+/datum/action/cooldown/mob_cooldown/mook_ability
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook.dm:260]: i think you should make a new file like `mook_abilities.dm` this file is too long. i see you're reliant on the defines you set in this file but if you either

A) make them global defines (preferred)
B) rename this file to `_mook.dm` and just remove the `#undef`s at the bottom

then the linter won't get mad at you

File: code/modules/mob/living/basic/lavaland/mook/mook_ai.dm [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+		finish_action(controller, FALSE)
+		return
+
+	controller.set_blackboard_key(village_key, home_marker)
+	finish_action(controller, TRUE)
+
+/datum/ai_behavior/wander
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION
+	var/wander_distance = 9
+	required_distance = 0
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook_ai.dm:133]: new vars after old vars.

File: code/modules/mob/living/basic/lavaland/mook/mook_village.dm [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+/obj/structure/material_stand/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	if(.)
+		return TRUE
+
+	if(!isliving(usr))
+		return TRUE
+
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/mook/mook_village.dm:73]:
```suggestion
	if(. || !isliving(usr))
		return TRUE
```

it's cleaner this way

File: config/lavaruinblacklist.txt [hunk 12]
```diff
@@ -39,4 +39,5 @@
 #_maps/RandomRuins/LavaRuins/lavaland_surface_elephant_graveyard.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_library.dmm
 #_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
-#_maps/RandomRuins/LavaRuins/lavaland_surface_phonebooth.dmm
\ No newline at end of file
+#_maps/RandomRuins/LavaRuins/lavaland_surface_phonebooth.dmm
+#_maps/RandomRuins/LavaRuins/lavaland_surface_mookvillage.dmm
```
Comment (san7890) [inline config/lavaruinblacklist.txt:43]: can you alphabetize this

**Review by san7890 — CHANGES_REQUESTED** [review]
enjoy more reviews

File: code/datums/ai/basic_mobs/basic_subtrees/mine_walls.dm (hunk 7 above)
Comment (Ben10Omintrix) [reply code/datums/ai/basic_mobs/basic_subtrees/mine_walls.dm:4]: it could work if the mob uses projectiles to mine, like the minebot

Comment (san7890) [reply code/datums/ai/basic_mobs/basic_subtrees/mine_walls.dm:4]: mkay

File: code/modules/mob/living/basic/lavaland/mook/mook.dm (hunk 9 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/lavaland/mook/mook.dm:256]: do u know where i can put these defines? i been looking for a home for them since i started this project

File: code/modules/mob/living/basic/lavaland/mook/mook.dm (hunk 8 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/lavaland/mook/mook.dm:243]: i tried this and the integration test cr*pped itself

File: code/modules/mob/living/basic/lavaland/mook/mook.dm (hunk 9 above)
Comment (san7890) [reply code/modules/mob/living/basic/lavaland/mook/mook.dm:256]: i just smack it somewhere in the mobs define file or make a new one in the folder

**Comment by gueba14** [comment]
This is wht i hope to see when i see pulls, new content thats not trinh to outhright brake/kill or replace something, love it

**Comment by san7890** [comment]
undraft when you're ready
